central/processlisteningonport/store/postgres/store.go

```diff
@@ -0,0 +1,763 @@
+}
+
+//// Stubs for satisfying legacy interfaces
+
+// AckKeysIndexed acknowledges the passed keys were indexed.
+func (s *storeImpl) AckKeysIndexed(ctx context.Context, keys ...string) error {
+	return nil
+}
+
+// GetKeysToIndex returns the keys that need to be indexed.
+func (s *storeImpl) GetKeysToIndex(ctx context.Context) ([]string, error) {
+	return nil, nil
+}
+
+//// Interface functions - END
+
+//// Used for testing
+
+// CreateTableAndNewStore returns a new Store instance for testing.
+func CreateTableAndNewStore(ctx context.Context, db *pgxpool.Pool, gormDB *gorm.DB) Store {
+	pkgSchema.ApplySchemaForTable(ctx, gormDB, baseTable)
+	return New(db)
+}
+
+// Destroy drops the tables associated with the target object type.
+func Destroy(ctx context.Context, db *pgxpool.Pool) {
+	dropTableProcessListeningOnPorts(ctx, db)
+}
+
+func dropTableProcessListeningOnPorts(ctx context.Context, db *pgxpool.Pool) {
+	_, _ = db.Exec(ctx, "DROP TABLE IF EXISTS process_listening_on_ports CASCADE")
+
+}
+
+//// Used for testing - END
+
+// Manually written function to get PLOP joined with ProcessIndicators
+func (s *storeImpl) GetProcessListeningOnPort(
+	ctx context.Context,
+	deploymentID string,
+) ([]*storage.ProcessListeningOnPort, error) {
+	defer metrics.SetPostgresOperationDurationTime(
+		time.Now(),
+		ops.GetProcessListeningOnPort,
+		"ProcessListeningOnPortStorage",
+	)
+
+	return pgutils.Retry2(func() ([]*storage.ProcessListeningOnPort, error) {
+		return s.retryableGetPLOP(ctx, deploymentID)
+	})
+}
+
+func (s *storeImpl) retryableGetPLOP(
+	ctx context.Context,
+	deploymentID string,
+) ([]*storage.ProcessListeningOnPort, error) {
+	var rows pgx.Rows
+	var err error
+
+	rows, err = s.db.Query(ctx, getByDeploymentStmt, deploymentID)
+
+	if err != nil {
+		log.Warnf("%s: %s", getByDeploymentStmt, err)
+		return nil, pgutils.ErrNilIfNoRows(err)
+	}
+	defer rows.Close()
+
+	return s.readRows(rows)
+}
+
+// Manual converting of raw data from SQL query to ProcessListeningOnPort (not
+// ProcessListeningOnPortStorage) object enriched with ProcessIndicator info.
+func (s *storeImpl) readRows(
+	rows pgx.Rows,
+) ([]*storage.ProcessListeningOnPort, error) {
+	var plops []*storage.ProcessListeningOnPort
+
+	for rows.Next() {
+		var id string
+		var serialized []byte
+		var podID string
+		var containerName string
+		var signalContainerID string
+		var signalName string
+		var signalArgs string
+		var signalExecFilePath string
+		var clusterID string
+		var procSerialized []byte
+
+		// We're getting ProcessIndicator directly from the SQL query, PLOP
+		// parts have to be extra deserialized.
+		if err := rows.Scan(
+			&id, &serialized,
+			&podID, &containerName,
+			&signalContainerID,
+			&signalName, &signalArgs, &signalExecFilePath,
+			&clusterID, &procSerialized); err != nil {
+			return nil, pgutils.ErrNilIfNoRows(err)
+		}
+
+		var msg storage.ProcessListeningOnPortStorage
+		if err := proto.Unmarshal(serialized, &msg); err != nil {
```

Comment:
If I remember correctly, this pattern was replaced in `pkg/search/postgres/common.go` to leverage the `Unmarshal` function in the generated code. The change was driven by performance considerations (https://github.com/stackrox/stackrox/pull/3451).
https://github.com/stackrox/stackrox/blob/master/pkg/search/postgres/common.go#L883-L893
I think the pattern here would be 
```
    msg := &storage.ProcessListeningOnPortStorage{}
    if err := msg.Unmarshal(serialized); err != nil {
        return nil, err
    }
```
This change is not critical in my eyes, applies to the ProcessIndicator decoding as well, and can be the target of a follow-up PR.